tools/internal_ci/linux/grpc_run_tests_matrix.sh

```diff
@@ -27,7 +27,8 @@ if [ -n "$KOKORO_GITHUB_PULL_REQUEST_NUMBER" ] && [ -n "$RUN_TESTS_FLAGS" ]; the
   export RUN_TESTS_FLAGS="$RUN_TESTS_FLAGS --filter_pr_tests --base_branch origin/$ghprbTargetBranch"
 fi
 
-tools/run_tests/run_tests_matrix.py $RUN_TESTS_FLAGS || FAILED="true"
+# Throttle stdout, so workers don't run out of memory
+tools/run_tests/run_tests_matrix.py $RUN_TESTS_FLAGS | pv -q -L 128000 || FAILED="true"
```

Comment:
Mixing pipes and `||` can be tricky (and depends on `pipefail` settings). I'd rather not make this change.
Also, AFAIK kokoro redirects both stdout and stderr to a .log file  and then the agent streams data from that file - so this change actually might not be helping with the problem at all (while making things complicated and more error prone on our side).